pkg/runtime/wfengine/workflow.go

```diff
@@ -209,6 +255,36 @@ func (wf *workflowActor) createWorkflowInstance(ctx context.Context, actorID str
 	return wf.saveInternalState(ctx, actorID, state)
 }
 
+// This method cleans up a workflow associated with the given actorID
+func (wf *workflowActor) cleanupOrchestrationStateInternal(ctx context.Context, actorID string, state *workflowState, requireCompleted bool) error {
+	runtimeState := getRuntimeState(actorID, state)
+	runtimeStatus := runtimeState.RuntimeStatus()
+	isCompleted := runtimeStatus == api.COMPLETED || runtimeStatus == api.TERMINATED || runtimeStatus == api.FAILED
```

Comment:
Can't we just use the existing `runtimeStatus.IsCompleted()` for this instead of checking specific status values? That would be more maintainable.

Reply:
The reason I am not using `runtimeStatus.IsCompleted()` is that I am not quite sure if it equals `[COMPLETED , TERMINATED , FAILED]`. From the code, the `completedEvent` is assigned when the orchestration completed at https://github.com/microsoft/durabletask-go/blob/aa335e20a192395649eb9ede97b57c9f647238ea/backend/runtimestate.go#L134, but when orchestration is terminated we don't assign a `completedEvent`, instead we create a `ExecutionTerminatedEvent` at https://github.com/microsoft/durabletask-go/blob/aa335e20a192395649eb9ede97b57c9f647238ea/backend/runtimestate.go#L197, so it seems `runtimeStatus.IsCompleted()` only check if orchestration is completed but not check if it's terminated/failed.

Reply:
According to [this integration test code](https://github.com/microsoft/durabletask-go/blob/aa335e20a192395649eb9ede97b57c9f647238ea/tests/orchestrations_test.go#L749-L757), a terminated orchestration results in `runtimeStatus.IsCompleted()` being `true`.

Reply:
Actually, hang on - I need to double check. The code I pointed to is for orchestration metadata, not the orchestration runtime state object...

Reply:
Okay, so confirming this wasn't as straightforward as I thought. But the short answer is that a terminated orchestration should always be completed. When an orchestration processes an ExecutionTerminated event, the SDK will generate a `completedEvent` with a status of "Terminated". This can be seen in the SDK code [here](https://github.com/microsoft/durabletask-go/blob/aa335e20a192395649eb9ede97b57c9f647238ea/task/orchestrator.go#L577-L579).

Reply:
Got it. Updated!